pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +119,253 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		}
+
+		var targetCluster string
+
+		// If a target cluster hasn't been selected, select one from the members
+		if instance.Spec.TargetCluster != "" {
+			targetCluster = instance.Spec.TargetCluster
+		} else {
+			// Automatic cluster selection
+			members := cluster.GetMemberClusters()
+			if len(members) > 0 {
+				targetCluster = members[0].Name
+			} else {
+				statusError := r.setStatusNoClustersAvailable(instance, "No member clusters found")
+				if statusError != nil {
+					reqLogger.Error(statusError, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+					return reconcile.Result{}, statusError
+				}
+
+				err = NewSignupError("No target clusters available")
+				reqLogger.Error(err, "No member clusters found")
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Provision the MasterUserRecord
+		err = r.provisionMasterUserRecord(instance, targetCluster, reqLogger)
+		if err != nil {
+			// If the error is because the MUR already exists, delay the next reconcile loop
+			if errors.IsAlreadyExists(err) {
+				return reconcile.Result{
+					RequeueAfter: 1 * time.Hour,
+				}, err
+			}
+
+			return reconcile.Result{}, err
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
+
+// provisionMasterUserRecord does the work of provisioning the MasterUserRecord
+func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1alpha1.UserSignup, targetCluster string, logger logr.Logger) error {
+	userAccounts := []toolchainv1alpha1.UserAccountEmbedded{
+		{
+			TargetCluster: targetCluster,
+			Spec: toolchainv1alpha1.UserAccountSpec{
+				NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+					Namespaces: []toolchainv1alpha1.Namespace{},
+				},
+			},
+		},
+	}
+
+	// TODO Update the MasterUserRecord with NSTemplateTier values
+	// SEE https://jira.coreos.com/browse/CRT-74
+
+	mur := &toolchainv1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userSignup.Spec.UserID,
+			Namespace: userSignup.Namespace,
+		},
+		Spec: toolchainv1alpha1.MasterUserRecordSpec{
+			UserID: userSignup.Spec.UserID,
+			UserAccounts: userAccounts,
+		},
+	}
+
+	err := controllerutil.SetControllerReference(mur, userSignup, r.scheme)
+	if err != nil {
+		logger.Error(err, "Error setting controller reference for MasterUserRecord %s", mur.Name)
+
+		statusError := r.setStatusFailedToCreateMUR(userSignup, "failed to set controller reference")
+		if statusError != nil {
+			logger.Error(statusError, "Error setting MUR failed status")
+			return statusError
+		}
+		return err
+	}
+
+	err = r.client.Create(context.TODO(), mur)
+	if err != nil {
+		// If the MasterUserRecord already exists (for whatever strange reason), we simply set the status to complete
+		if errors.IsAlreadyExists(err) {
+			statusError := r.setStatusComplete(userSignup, "")
```

Comment:
Maybe worth adding a warning. It's OK if the MUR already exists. We can handle it. But for monitoring purposes it's worth log it.

Reply:
Added Info() message